lib/Targets/AIETargetXAIEV1.cpp

```diff
@@ -279,14 +279,26 @@ mlir::LogicalResult AIETranslateToXAIEV1(ModuleOp module, raw_ostream &output) {
       int acqValue = 0, relValue = 0;
       StringRef acqEnable = disable;
       StringRef relEnable = disable;
-      int lockID;
+      int lockID = -1;
       for (auto op : block.getOps<UseLockOp>()) {
         LockOp lock = dyn_cast<LockOp>(op.lock().getDefiningOp());
+        if (lockID != -1 && lockID != lock.getLockID()) {
+          op->emitError("Each DmaBd can use up to one lock.");
+          return failure();
+        }
```

Comment:
Seems like these should be verifer conditions?

Reply:
Done.